### PR TITLE
Rationalize embedder.dart and shell.dart

### LINKED ISSUE #1803: Attempting to connect to a non-existent mojo service in skyshell throws exception
```dart
10-26 14:08:51.332 I/sky     (10760): The null object does not have a getter 'ptr'.
10-26 14:08:51.332 I/sky     (10760): NoSuchMethodError: method not found: 'ptr'
10-26 14:08:51.332 I/sky     (10760): Receiver: null
10-26 14:08:51.332 I/sky     (10760): Arguments: []
10-26 14:08:51.332 I/sky     (10760): Stack trace:
10-26 14:08:51.592 I/sky     (10760): #0      Object._noSuchMethod (dart:core-patch/object_patch.dart:42)
10-26 14:08:51.592 I/sky     (10760): #1      Object.noSuchMethod (dart:core-patch/object_patch.dart:45)
10-26 14:08:51.592 I/sky     (10760): #2      _EmbedderImpl.connectToService (http://localhost:9888/packages/flutter/src/services/embedder.dart:58:11)
```

We should probably do something more thoughtful in this situation.

### PR DESCRIPTION
**Pull request by abarth**
This patch combines embedder.dart and shell.dart into one thing. We should now
handle a bunch of error cases better.

  * embedder.connectToApplication has moved to shell.connectToApplication,
    matching the rest of the mojo universe.
  * embedder.connecttoService has moved to shell.connnectToService (and merged
    with shell.requestService).
  * shell.requestService is now shell.connectToService, matching the rest of
    the mojo universe.
  * serviceRegistry has moved from embedder.serviceRegistry to a top-level
    getter.

Fixes #1803

### CODE REVIEW

**Comment by Hixie** [comment]
LGTM but @jamesr would be a better reviewer.

File: sky/packages/sky/lib/src/services/service_registry.dart [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+ServiceRegistryProxy _initServiceRegistryProxy() {
+  core.MojoHandle serviceRegistryHandle = new core.MojoHandle(internals.takeShellProxyHandle());
+  if (!serviceRegistryHandle.isValid)
+    return null;
+  return new ServiceRegistryProxy.fromHandle(serviceRegistryHandle);
+}
+
+final ServiceRegistryProxy _serviceRegistryProxy = _initServiceRegistryProxy();
+final ServiceRegistry serviceRegistry = _serviceRegistryProxy?.ptr;
```
Comment (jamesr) [inline sky/packages/sky/lib/src/services/service_registry.dart]: what does '?.ptr' mean?

Comment (abarth) [inline sky/packages/sky/lib/src/services/service_registry.dart]: It's equivalent to:

`_serviceRegistryProxy ==  null ? null : _serviceRegistryProxy.ptr`

**Comment by jamesr** [comment]
lgtm

**Pull request merged**